test/metrics/index.test.js

```diff
@@ -211,6 +211,18 @@ test('noop counter when not active', function (t) {
   t.end()
 })
 
+test('Metrics objects do not throw/crash when not started', function (t) {
```

Comment:
nit: Could perhaps add a test between these two lines to test `agent.registerMetric(...)` before `agent.start() is called: https://github.com/elastic/apm-agent-nodejs/blob/22423a5b409b8af0a9923624e9af32e591a8159e/test/agent.test.js#L1423-L1424

Reply:
Hmmm -- good suggestion but after looking into it my current thinking is to hold off on that -- I'm not sure that additional call brings much value to the test suite.  The `_metrics` object is mocked in that test so we're not actually hitting any of the code paths that were a problem.   Also, it's not clear what we could assert on there given that `registerMetrics` returns nothing regardless of whether the metric is created successfully or not. 

Unless you think changing `registerMetric` to return the value of the call to `this._metrics.getOrCreateGauge(name, callback, labels)` is worthwhile and not a breaking change? @trentm

Reply:
> The _metrics object is mocked in that test so we're not actually hitting any of the code paths that were a problem. 

You had me at "this is a mock".